### PR TITLE
Staging run

### PR DESCRIPTION
**Pull request by ludwikkazmierczak**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 1]
```diff
@@ -50,22 +63,55 @@ public String getUrlForPath(String wikiPath) {
...
+        String devBoxOwner = this.env.split("-")[1];
+        return String.format(DEV_URL_FORMAT, www, wikiaName, devBoxOwner, env.getWikiaDomain());
+      }
+      case PROD: {
+        return String.format(PROD_URL_FORMAT, www, wikiaName, env.getWikiaDomain());
+      }
+      case STAGING: {
+        return String.format(PROD_URL_FORMAT, www, wikiaName, env.getWikiaDomain());
+      }
+      case SANDBOX: {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 2]
```diff
@@ -50,22 +63,55 @@ public String getUrlForPath(String wikiPath) {
...
+    final String wikiaName = isGlobal(wikiName);
+
+    String www = "";
+    if (addWWW) {
+      www = "www.";
     }
 
-    return composeUrl(prefix, wikiName, suffix);
+    switch (env) {
+      case DEV: {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 3]
```diff
@@ -50,22 +63,55 @@ public String getUrlForPath(String wikiPath) {
...
+        String devBoxOwner = this.env.split("-")[1];
+        return String.format(DEV_URL_FORMAT, www, wikiaName, devBoxOwner, env.getWikiaDomain());
+      }
+      case PROD: {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 4]
```diff
@@ -50,22 +63,55 @@ public String getUrlForPath(String wikiPath) {
...
-    if (env.contains("dev") && !env.contains(SANDBOX_MERCURY_DEV)) {
-      String devBoxOwner = env.split("-")[1];
+  public String getUrlForWiki(boolean addWWW) {
+    return getUrlForWiki(Configuration.getWikiName(), addWWW);
+  }
 
-      suffix = "." + devBoxOwner + "." + "wikia-dev.com";
-    } else {
-      suffix = wikiName.endsWith("wikia") ? ".com" : ".wikia.com";
+  public String getUrlForWiki(String wikiName, boolean addWWW) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR The Cyclomatic Complexity of this method "getUrlForWiki" is 11 which is greater than 10 authorized. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 5]
```diff
@@ -50,22 +63,55 @@ public String getUrlForPath(String wikiPath) {
...
-    return composeUrl(prefix, wikiName, suffix);
+    switch (env) {
+      case DEV: {
+        String devBoxOwner = this.env.split("-")[1];
+        return String.format(DEV_URL_FORMAT, www, wikiaName, devBoxOwner, env.getWikiaDomain());
+      }
+      case PROD: {
+        return String.format(PROD_URL_FORMAT, www, wikiaName, env.getWikiaDomain());
+      }
+      case STAGING: {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Extract this nested code block into a method. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)

File: src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java [hunk 6]
```diff
@@ -50,22 +63,55 @@ public String getUrlForPath(String wikiPath) {
...
-      String devBoxOwner = env.split("-")[1];
+  public String getUrlForWiki(boolean addWWW) {
+    return getUrlForWiki(Configuration.getWikiName(), addWWW);
+  }
 
-      suffix = "." + devBoxOwner + "." + "wikia-dev.com";
-    } else {
-      suffix = wikiName.endsWith("wikia") ? ".com" : ".wikia.com";
+  public String getUrlForWiki(String wikiName, boolean addWWW) {
+    EnvType env = Configuration.getEnvType(this.env);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java]: MAJOR Rename "env" which hides the field declared at line 12. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: src/test/java/com/wikia/webdriver/common/core/configuration/Configuration.java [hunk 7]
```diff
@@ -141,32 +146,40 @@ public static String getEmulator() {
...
-    return "";
-  }
-
-  public static String getWikiaDomain() {
-    if (StringUtils.isBlank(wikiaDomain)) {
-      wikiaDomain = "dev".equals(getEnvType()) ? ".wikia-dev.com" : ".wikia.com";
+    return EnvType.PROD;
+  }
+
+  public static EnvType getEnvType(String env) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/configuration/Configuration.java]: MAJOR The Cyclomatic Complexity of this method "getEnvType" is 11 which is greater than 10 authorized. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/test/java/com/wikia/webdriver/common/core/configuration/Configuration.java [hunk 8]
```diff
@@ -17,45 +9,58 @@
...
 
   private static final String DEFAULT_CONFIG_FILE_NAME = "config_default.yml";
   private static final String LOCAL_CONFIG_FILE_NAME = "config.yml";
   private static final Logger LOGGER = Logger.getLogger(Configuration.class.getName());
-  private static String wikiaDomain;
   private static String geoEdgeCountry;
   private static Map<String, String> defaultConfig;
   private static Map<String, String> testConfig = new HashMap<>();
+  @Getter(lazy = true)
+  private static final String wikiaDomain = getEnvType().getWikiaDomain();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/configuration/Configuration.java]: MAJOR Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00115)

File: src/test/java/com/wikia/webdriver/common/core/configuration/Configuration.java [hunk 9]
```diff
@@ -141,32 +146,40 @@ public static String getEmulator() {
...
+  public static String getPageLoadStrategy() {
     return getProp("unstablePageLoadStrategy");
   }
 
   public static Credentials getCredentials() {
     return new Credentials();
   }
 
-  public static String getEnvType() {
+  public static EnvType getEnvType() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/configuration/Configuration.java]: MAJOR The Cyclomatic Complexity of this method "getEnvType" is 11 which is greater than 10 authorized. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues:
* MAJOR 9 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'com.wikia.webdriver.common.core.url.UrlBuilder'.](https://github.com/Wikia/selenium-tests/blob/8b33f013e3261dd0903f0328650b9b276af5b2bf/src/test/java/com/wikia/webdriver/elements/oasis/pages/ArticlePreviewPage.java#L3) [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Remove this unused import 'com.wikia.webdriver.common.core.url.UrlBuilder'.](https://github.com/Wikia/selenium-tests/blob/8b33f013e3261dd0903f0328650b9b276af5b2bf/src/test/java/com/wikia/webdriver/testcases/mercurytests/old/curatedcontenttests/NavigationTests.java#L13) [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java [hunk 10]
```diff
@@ -9,6 +9,7 @@
 import com.wikia.webdriver.common.core.Helios;
 import com.wikia.webdriver.common.core.MailFunctions;
 import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.core.configuration.EnvType;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.configuration.EnvType'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues:
* MAJOR 9 major
* MINOR 3 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'com.wikia.webdriver.common.core.url.UrlBuilder'.](https://github.com/Wikia/selenium-tests/blob/a344f08a472d1c1e764fa9963611d9eabf3abd71/src/test/java/com/wikia/webdriver/elements/oasis/pages/ArticlePreviewPage.java#L3) [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)
* MINOR [Remove this unused import 'com.wikia.webdriver.common.core.url.UrlBuilder'.](https://github.com/Wikia/selenium-tests/blob/a344f08a472d1c1e764fa9963611d9eabf3abd71/src/test/java/com/wikia/webdriver/testcases/mercurytests/old/curatedcontenttests/NavigationTests.java#L13) [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by nikodamn** [comment]
LGTM

**Pull request merged**